pkg/ottl/ottlfuncs/func_replace_all_matches_test.go

```diff
@@ -30,8 +31,33 @@ func Test_replaceAllMatches(t *testing.T) {
 		target      ottl.PMapGetter[pcommon.Map]
 		pattern     string
 		replacement ottl.StringGetter[pcommon.Map]
+		function    ottl.Optional[ottl.FunctionGetter[pcommon.Map]]
 		want        func(pcommon.Map)
 	}{
+		{
+			name:    "replace only matches (with hash function)",
+			target:  target,
+			pattern: "hello*",
+			replacement: ottl.StandardStringGetter[pcommon.Map]{
+				Getter: func(context.Context, pcommon.Map) (interface{}, error) {
+					return "hello {universe}", nil
+				},
+			},
+			function: ottl.Optional[ottl.FunctionGetter[pcommon.Map]]{
+				Val: ottl.StandardFunctionGetter[pcommon.Map]{
+					FCtx: ottl.FunctionContext{
+						Set: componenttest.NewNopTelemetrySettings(),
+					},
+					Fact: StandardConverters[pcommon.Map]()["SHA256"],
+				},
+				HasValue: true,
+			},
```

Comment:
We should use `ottl.NewTestingOptional`

Reply:
thanks will take a look at it and make the updates.

Reply:
I've updated the PR to use `ottl.NewTestingOptional`